lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -98,11 +98,16 @@ sub _poll_cache_service ($job, $cache_client, $request, $delay, $callback) {
 
 sub cache_assets ($cache_client, $job, $vars, $assets_to_cache, $assetkeys, $webui_host, $pooldir, $callback) {
     return $callback->(undef) unless my $this_asset = shift @$assets_to_cache;
-    return cache_assets(@_) unless my $asset_value = $vars->{$this_asset};
+    my $asset_value = $vars->{$this_asset};
+    unless ($asset_value) {
+        return cache_assets($cache_client, $job, $vars, $assets_to_cache, $assetkeys, $webui_host, $pooldir, $callback);
+    }
```

Comment:
not sure why you turned that around. Especially with "unless" we commonly prefer the post-condition form

```suggestion
    return cache_assets($cache_client, $job, $vars, $assets_to_cache, $assetkeys, $webui_host, $pooldir, $callback) unless $asset_value;
```

Reply:
Because it makes the lines super long. But if you don't mind that, sure.